### PR TITLE
OpenFileModal, NewItemModal and Message Modal.

### PR DESCRIPTION
**Pull request by theXnoob**

### CODE REVIEW

**Comment by Gaztin** [comment]
Looks like you have conflicts. I assume it's because of the PNG files in /resources/

Try doing a `git rebase -r upstream master` and follow by force-pushing to this branch

**Review by Gaztin — CHANGES_REQUESTED** [review]

File: src/Geno/C++/GUI/Modals/IModal.h [hunk 1]
```diff
@@ -50,6 +52,11 @@ class IModal
 
 //////////////////////////////////////////////////////////////////////////
 
+	ImGuiAux::ButtonData m_ButtonData = {};
+
+	ImVec2 m_MinSize = ImVec2( 350, 196 );
+	ImVec2 m_MaxSize = ImVec2( 550, 396 );
+
```
[the remark this answers is not in the document]
Comment (theXnoob) [reply src/Geno/C++/GUI/Modals/IModal.h:59]: yeah so we can set it in loop also

[the remark this answers is not in the document]
Comment (Gaztin) [reply src/Geno/C++/GUI/Modals/IModal.h:59]: what loop?

[the remark this answers is not in the document]
Comment (theXnoob) [reply src/Geno/C++/GUI/Modals/IModal.h:59]: i mean the IModal::Update() function which will run into loop

[the remark this answers is not in the document]
Comment (Gaztin) [reply src/Geno/C++/GUI/Modals/IModal.h:59]: But if it's in IModal::Update(), then it has access to its own private members

[the remark this answers is not in the document]
Comment (theXnoob) [reply src/Geno/C++/GUI/Modals/IModal.h:59]: IModal::Update() call UpdateDerived() so we can set that in UpdateDerived()

[the remark this answers is not in the document]
Comment (Gaztin) [reply src/Geno/C++/GUI/Modals/IModal.h:59]: Ah. Now I see what you mean

File: src/Geno/C++/Auxiliary/ImGuiAux.cpp [hunk 2]
```diff
@@ -24,15 +24,9 @@
 
 //////////////////////////////////////////////////////////////////////////
 
-bool ImGuiAux::RenameTree( std::string& rNameToRename )
+void ImGuiAux::RenameTree( std::string& rNameToRename, bool& rRename, const std::function< bool( void ) >& rCallback )
```
Comment (Gaztin) [inline src/Geno/C++/Auxiliary/ImGuiAux.cpp:27]: Can't this also just return a bool, like it did before? Or is there some other reason as to why this was changed?

File: src/Geno/C++/GUI/Modals/IModal.h [hunk 3]
```diff
@@ -20,6 +20,8 @@
 
 #include <Common/Macros.h>
 
+#include "Auxiliary/ImGuiAux.h"
+
```
Comment (Gaztin) [inline src/Geno/C++/GUI/Modals/IModal.h:24]: Internal includes before external includes

File: src/Geno/C++/GUI/Modals/NewItemModal.cpp [hunk 4]
```diff
@@ -17,97 +17,114 @@
...
 	if( Open() )
 	{
-		m_Title       = std::move( Title );
-		m_Location    = std::move( DefaultLocation );
-		m_Callback    = reinterpret_cast< void* >( Callback );
-		m_pUser       = pUser;
-		m_RequestType = RequestTypePath;
-	}
+		m_Title    = Title;
+		m_pFilter  = pFilter;
```
Comment (Gaztin) [inline src/Geno/C++/GUI/Modals/NewItemModal.cpp:42]: Looks kinda dangerous to store a copy of the char*. Do we even need to store it? It seems to only be used inside this function anyway

File: src/Geno/C++/GUI/Modals/OpenFileModal.cpp [hunk 5]
```diff
@@ -17,335 +17,416 @@
...
+									continue;
+							}
 
-							// Open file immediately if file was double-clicked
-							if( ImGui::IsMouseDoubleClicked( ImGuiMouseButton_Left ) )
+							if( m_SearchEnabled && !m_SearchResult.empty() )
 							{
-								if( m_Callback )
-									m_Callback( m_SelectedPath, m_pUser );
+								if( !strstr( Path.filename().string().c_str(), m_SearchResult.c_str() ) )
```
Comment (Gaztin) [inline src/Geno/C++/GUI/Modals/OpenFileModal.cpp:293]: You can do this without C functions by using `std::string::find()`

**Review by Gaztin — CHANGES_REQUESTED** [review]

File: src/Geno/C++/GUI/Modals/NewItemModal.cpp (hunk 4 above)
Comment (theXnoob) [reply src/Geno/C++/GUI/Modals/NewItemModal.cpp:42]: i will change this

File: src/Geno/C++/GUI/Modals/IModal.h (hunk 3 above)
Comment (theXnoob) [reply src/Geno/C++/GUI/Modals/IModal.h:24]: will fix it

File: src/Geno/C++/GUI/Modals/OpenFileModal.cpp (hunk 5 above)
Comment (Gaztin) [reply src/Geno/C++/GUI/Modals/OpenFileModal.cpp:293]: It's just generally better to use C++ functions over C functions, especially if we already have the structure (in this case std::string) that we need.
But this is fine for now

**Review by Gaztin — APPROVED** [review]

**Pull request merged**